scripts/needs_testrun.py

```diff
@@ -58,43 +58,36 @@ def get_merge_base(pr_number: int) -> str:
 
 
 @cache
-def get_changed_files(pr_number: int) -> t.Set[str]:
+def get_changed_files(pr_number: int, sha: t.Optional[str] = None) -> t.Set[str]:
     """Get the files changed in a PR
 
+    Try with the GitHub REST API for the most accurate result. If that fails,
+    or if there is a specific SHA given, use the less accurate method of
+    diffing against a base commit, either the given SHA or the merge-base.
+
     >>> sorted(get_changed_files(6388))  # doctest: +NORMALIZE_WHITESPACE
     ['ddtrace/debugging/_expressions.py',
     'releasenotes/notes/fix-debugger-expressions-none-literal-30f3328d2e386f40.yaml',
     'tests/debugging/test_expressions.py']
     """
-    try:
-        # Try with the GitHub REST API for the most accurate result
-        url = f"https://api.github.com/repos/datadog/dd-trace-py/pulls/{pr_number}/files"
-        headers = {"Accept": "application/vnd.github+json"}
-
-        return {_["filename"] for _ in json.load(urlopen(Request(url, headers=headers)))}
-
-    except Exception:
-        # If that fails use the less accurate method of diffing against the
-        # merge-base w.r.t. the base branch
-        LOGGER.warning("Failed to get changed files from GitHub API, using git diff instead")
-        return set(
-            check_output(
-                [
-                    "git",
-                    "diff",
-                    "--name-only",
-                    "HEAD",
-                    get_merge_base(pr_number),
-                ]
-            )
-            .decode("utf-8")
-            .strip()
-            .splitlines()
-        )
+    rest_check_failed = False
+    if sha is None:
+        try:
+            url = f"https://api.github.com/repos/datadog/dd-trace-py/pulls/{pr_number}/files"
+            headers = {"Accept": "application/vnd.github+json"}
+            return {_["filename"] for _ in json.load(urlopen(Request(url, headers=headers)))}
+        except Exception:
```

Comment:
It might be helpful to log the exception. This can help us debug/monitor different types of errors.